src/main/java/com/ibm/crypto/plus/provider/XDHPublicKeyImpl.java

```diff
@@ -411,7 +411,9 @@ public byte[] getEncoded() {
                 oidSeq.putOID(this.algid.getOID());
                 if ((oidSubSeq != null)) {
                     oidSeq.write(DerValue.tag_Sequence, oidSubSeq);
-                } else {
+                }
+                
+                if (Integer.parseInt(provider.getVersionStr()) <= 11) {
```

Comment:
The provider version comes from `System.getProperty("java.specification.version")` in `OpenJCEPlusProvider` class, link below. When initializing the `OpenJCEPlus` or `OpenJCEPlusFIPS` provider, there is no constructor to specify the provider version, the version will always come from the `java.specification.version`. So, the provider version should always match the Java version.

https://github.com/IBM/OpenJCEPlus/blob/main/src/main/java/com/ibm/crypto/plus/provider/OpenJCEPlusProvider.java#L25

Reply:
Added a new method `getJavaVersionStr(`) in `OpenJCEPlusProvider`, to return the `java.specification.version` in case the provider version may different with the Java version in the future.